libstuff/SQResult.cpp

```diff
@@ -183,3 +188,55 @@ SQResult& SQResult::operator=(const SQResult& other) {
     }
     return *this;
 }
+
+vector<SQResultRow>::iterator SQResult::begin() {
+    return rows.begin();
+}
+
+vector<SQResultRow>::iterator SQResult::end() {
+    return rows.end();
+}
+
+vector<SQResultRow>::const_iterator SQResult::begin() const {
+    return rows.begin();
+}
+
+vector<SQResultRow>::const_iterator SQResult::end() const {
+    return rows.end();
+}
+
+vector<SQResultRow>::const_iterator SQResult::cbegin() const {
+    return rows.cbegin();
+}
+
+vector<SQResultRow>::const_iterator SQResult::cend() const {
+    return rows.cend();
+}
+
+void SQResult::resize(size_t newSize) {
+    rows.resize(newSize);
+    for (auto& row : rows) {
+        row.result = this;
+    }
+}
+
+void SQResult::push_back(const SQResultRow& row) {
+    SQResultRow newRow = row;
+    newRow.result = this;
```

Comment:
We don't need the two lines above, given the existing (and very limited) use here. We expect `row` to already have the correct value set for `result`.